WooCommerce/Classes/ViewRelated/Orders/Collect Payments/LegacyCollectOrderPaymentUseCase.swift

```diff
@@ -172,6 +172,7 @@ final class LegacyCollectOrderPaymentUseCase: NSObject, LegacyCollectOrderPaymen
                 self.trackPaymentCancelation()
                 onCancel()
             case .failure(let error):
+                CardPresentPaymentOnboardingStateCache.shared.invalidate()
```

Comment:
This may also need to be done in the [client-side refund code](https://github.com/woocommerce/woocommerce-ios/blob/52964209552fa549e541b0eda7591a6d606a10d0/WooCommerce/Classes/ViewRelated/Orders/Refund/RefundSubmissionUseCase.swift#L158) for Interac refunds

Reply:
Good hint, fixed in af42dbfebb2ba5183907040338f78294211e7e09